Simplified/NYPLBookRegistry.h

```diff
@@ -13,7 +13,25 @@ static NSString *const _Nonnull NYPLBookRegistryDidChangeNotification =
 static NSString *const _Nonnull NYPLBookProcessingDidChangeNotification =
   @"NYPLBookProcessingDidChangeNotification";
 
-@interface NYPLBookRegistry : NSObject
+@protocol NYPLBookRegistryProvider <NSObject>
+
+- (nonnull NSArray<NYPLReadiumBookmark *> *)readiumBookmarksForIdentifier:(nonnull NSString *)identifier;
```

Comment:
Thanks for adding this protocol. One step forward toward less tight coupling 👍 

I think the return value for this and `locationForIdentifier:` should be nullable though... what If I pass in an invalid identifier?

Secondly, are all these apis implemented in Objc? if some of them are implemented in swift, nonnulls can lead to a crash if the argument is actually nil.

Reply:
`readiumBookmarksForIdentifier:` returns an empty array if invalid identifier being passed in.

`locationForIdentifier:` is already nullable.